tests/unit/test_dataset.py

```diff
@@ -92,7 +93,7 @@ def test_const_dataset__conflicting_data():
 
 
 def test_const_dataset__single_data(dataset_type):
-    components = {"node": 3, "sym_load": 2, "asym_load": 4}
+    components = {ComponentType.node: 3, ComponentType.sym_load: 2, ComponentType.asym_load: 4}
```

Comment:
A bit awkward if I have to be brutally honest about the new usage... I wouldn't be surprised if users would 1, stick to the old way, or 2, come up with ways like:
 ```suggestion
    def map_components(enum_class, simple_mapping):
        return {getattr(enum_class, k): v for k, v in simple_mapping.items()}
    
    components = map_components(ComponentType, {
        'node': 3,
        'sym_load': 2,
        'asym_load': 4
    })
```
or plainly:
```suggestion
    temp_mapping = {
        'node': 3,
        'sym_load': 2,
        'asym_load': 4
    }
    
    components = {getattr(ComponentType, k): v for k, v in temp_mapping.items()}
```

Reply:
@Jerry-Jinfeng-Guo But the good thing is that support is there for the new and old implementations. Users can do `ComponentType.node: 3` or `"node": 3`, and it is going to be treated equally within pgm. However, now they will have proper warnings from linters if they make typos and use the new implementation. It is a matter of perspective.